src/hooks/usePaginatedMovies.js

```diff
@@ -0,0 +1,26 @@
+// src/hooks/usePaginatedMovies.js
+import { useState, useEffect } from 'react';
+
+const usePaginatedMovies = (page, limit) => {
+  const [records, setRecords] = useState([]);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchMovies = async () => {
```

Comment:
Let's discuss the hooks design together. Especially we can group these hooks into a single file.